sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/implementation/ModelsQuery.java

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.iot.modelsrepository.implementation;
+
+import com.azure.iot.modelsrepository.implementation.models.ModelMetadata;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The {@link ModelsQuery} class is responsible for parsing DTDL v2 models to produce key metadata.
+ * In the current form this class is focused on determining model dependencies recursively
+ * via extends and component schemas.
+ */
+public class ModelsQuery {
+
+    private static final String UTF8_BOM = "\uFEFF";
+    private final String content;
+    private final ObjectMapper mapper;
+
+    public ModelsQuery(String content) {
+        mapper = new ObjectMapper();
+        if (content.startsWith(UTF8_BOM)) {
+            this.content = content.substring(1);
+        } else {
+            this.content = content;
+        }
+    }
+
+    private static ModelMetadata parseInterface(JsonNode root) {
+        String rootDtmi = parseRootDtmi(root);
```

Comment:
do you need to check if root.isObject here?

Reply:
parseRootDtmi does that check.